src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -1,21 +1,30 @@
-import { FC } from "react";
+import { FC, useEffect, useState } from "react";
 
 import { Track, Progress } from "./ProgressBar.styles";
+import getProgressBarAnimationProps from "./helpers/getProgressBarAnimationProps";
 
 const ProgressBar: FC<{
   startTime: number;
   endTime: number;
 }> = ({ startTime, endTime }) => {
-  const now = Date.now();
-  const done = now - startTime;
-  const left = endTime - startTime;
-  const fractionDone = done / left;
+  const [isMounted, setIsMounted] = useState(false);
+
+  useEffect(() => {
+    setIsMounted(true);
```

Comment:
Took me a minute to understand why this was needed. Might be worth adding a comment e.g.:

```
// `isMounted` will be false until the component has rendered once. We use this
// so that we can set the initial width of the progress bar to correspond with
// the initial progress, before setting the "target" to 100% for the CSS transition
```

Reply:
Sharp!